src/python/pants/engine/internals/graph.py

```diff
@@ -1288,15 +1296,20 @@ async def resolve_dependencies(
     local_environment_name: ChosenLocalEnvironmentName,
 ) -> Addresses:
     environment_name = local_environment_name.val
-    wrapped_tgt, explicitly_provided = await MultiGet(
-        Get(
-            WrappedTarget,
-            # It's only possible to find dependencies for a target that we already know exists.
-            WrappedTargetRequest(request.field.address, description_of_origin="<infallible>"),
-        ),
-        Get(ExplicitlyProvidedDependencies, DependenciesRequest, request),
+    wrapped_tgt = await Get(
+        WrappedTarget,
+        # It's only possible to find dependencies for a target that we already know exists.
+        WrappedTargetRequest(request.field.address, description_of_origin="<infallible>"),
     )
     tgt = wrapped_tgt.target
+    try:
+        explicitly_provided = await Get(
+            ExplicitlyProvidedDependencies, DependenciesRequest, request
+        )
+    except Exception as e:
+        raise InvalidFieldException(
+            f"{tgt.description_of_origin}: Failed to get dependencies for {tgt.address}: {e}"
```

Comment:
Nit: Maybe...:?
```suggestion
            f"Failed to get dependencies for {tgt.address} (defined by {tgt.description_of_origin}): {e}"
```

Alternatively, maybe give `InvalidFieldException` the `description_of_origin` as an argument, and then render the full message in its `str`/`repr` to normalize how it is rendered in all instances? Ditto other target exceptions.